src/weekly-notes/index.js

```diff
@@ -161,10 +159,11 @@ function getCurrentWeek() {
 }
 
 function getIssueTitle() {
-  const currentWeek = getCurrentWeek(),
+  const weekInterval = core.getInput('week-interval'),
+        currentWeek = getCurrentWeek(),
         currentWeekNr = currentWeek.weekNumber,
         currentYearNr = currentWeek.year,
-        upcomingWeekNr = currentWeekNr == 52 ? 1 : currentWeekNr + weekInterval,
+        upcomingWeekNr = (currentWeekNr + weekInterval - 1) % 52 + 1,
```

Comment:
@nikku Assuming we are on the last week of the year and we have an interval of 2 weeks we would have `(53+2-1)%52+1` which is equals `3`.

Don't we want the result to be `2`? first week from W53 to W1 and and second week from W1 to W2

Reply:
I guess I'm mixing it up because W53===W1, right? so it makes sense for it to be 3

Reply:
Last week of the year is `52`, not `53`, `currentWeekNr` is in range `1..52`. Now assuming `currentWeekNr=52`, `weekInterval=2` I believe we want to end up with `upcomingWeekNr=2`:

```
(52 + 2 - 1) % 52 + 1 = x
(54 - 1) % 52 + 1 = x
(53) % 52 + 1 = x
1 + 1 = x
2 = x
```

Does this make sense?

Reply:
https://github.com/bpmn-io/actions/pull/17#discussion_r1617934575 is not fully correct.

Last week of the year is 53 for a leap year (one day), currentWeekNr is in range 1..53. Now assuming currentWeekNr=53 (leap year), weekInterval=2, what will we end up with?

```
(53 + 2 - 1) % 52 + 1 = x
(55 - 1) % 52 + 1 = x
(54) % 52 + 1 = x
2 + 1 = x
3 = x
```

In the situation of a leap year the next weekly will be in week `3`. We kept the existing bug. :slightly_smiling_face: 

If we wanted to fix this, then we'd want to account for max `52` weeks, regardless of leap year:

```suggestion
        upcomingWeekNr = (Math.min(currentWeekNr, 52) + weekInterval - 1) % 52 + 1,
```

This would yield `W2` in the next year again, also for `currentWeekNr=53`.

Reply:
Proposal: introduce tests for the action.

Reply:
@barmac @nikku I'll handle this before lunch and I'll try to a couple of tests

Reply:
@vsgoulart Already did it! Please check if it works for you.